solidity/contracts/relay/RelayMaintainerProxy.sol

```diff
@@ -0,0 +1,100 @@
+// SPDX-License-Identifier: GPL-3.0-only
+
+// ██████████████     ▐████▌     ██████████████
+// ██████████████     ▐████▌     ██████████████
+//               ▐████▌    ▐████▌
+//               ▐████▌    ▐████▌
+// ██████████████     ▐████▌     ██████████████
+// ██████████████     ▐████▌     ██████████████
+//               ▐████▌    ▐████▌
+//               ▐████▌    ▐████▌
+//               ▐████▌    ▐████▌
+//               ▐████▌    ▐████▌
+//               ▐████▌    ▐████▌
+//               ▐████▌    ▐████▌
+
+pragma solidity 0.8.17;
+
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "@keep-network/random-beacon/contracts/Reimbursable.sol";
+import "@keep-network/random-beacon/contracts/ReimbursementPool.sol";
+
+import "./LightRelay.sol";
+
+/// @title LightRelayMaintainerProxy
+/// @notice The proxy contract that allows the relay maintainers to be refunded
+///         for the spent gas from the `ReimbursementPool`. When proving the
+///         next Bitcoin difficulty epoch, the maintainer calls the
+///         `LightRelayMaintainerProxy` which in turn calls the actual `LightRelay`
+///         contract.
+contract LightRelayMaintainerProxy is Ownable, Reimbursable {
+    ILightRelay public lightRelay;
+
+    /// @notice Stores addresses that can maintain the relay.
```

Comment:
Let's be more explicit why this is required:

```suggestion
    /// @notice Stores the addresses that can maintain the relay. Those
    ///         addresses are attested by the DAO.
    /// @dev The goal is to prevent a griefing attack by frontrunning relay
    ///      maintainer which is responsible for retargeting the relay in the
    ///      given round. The maintainer's transaction would revert with no gas
    ///      refund. Having the ability to restrict maintainer addresses is also
    ///      important in case the underlying relay contract has authorization
    ///      requirements for callers.
```

Reply:
Done in 52725659de211ab6a316eafe1b672ed61b0e06d1.